### DISCUSSION TITLE
Is there a way to retrieve the path of Node used by VS Code or a way to execute a node script using code binary?

### DISCUSSION THREAD

**Discussion by apupier** [microsoft/vscode-discussions#746] [Extension Development QnA] [answered]
Currently, we are using a library which is calling `process.execPath` variable to retrieve the path to `node` binary used to execute the current program.

In VS Code context, this variable is not returning `node` binary path but `code` binary path.

See same question on Stack Overflow (unfortunately without answers from someone else several months ago) https://stackoverflow.com/questions/75103016/node-execpath-references-vscode-installation-instead-of-path-to-node-installatio and the specific context where I'm hitting the issue: https://github.com/asciidoctor/asciidoctor-vscode/pull/737#issuecomment-1626956520

**Comment by isidorn** [comment]
VS Code uses node that is coming with Electron.
So we do not rely on outside Node afaik.

**Comment by r3m0t** [comment, marked as answer]
If the subprocess has `ELECTRON_RUN_AS_NODE` set, it should behave like a normal Node.js process, accepting a `.js` path and etc. https://www.electronjs.org/docs/latest/api/environment-variables
